lib/bolt/inventory/inventory2.rb

```diff
@@ -157,23 +125,23 @@ def update_target(target)
           data['config'] = {}
         end
 
-        data = Bolt::Inventory.localhost_defaults(data) if target.name == 'localhost'
-        # These should only get set from the inventory if they have not yet
-        # been instantiated
-        set_vars_from_hash(target.name, data['vars']) unless @target_vars[target.name]
-        set_facts(target.name, data['facts']) unless @target_facts[target.name]
-        data['features']&.each { |feature| set_feature(target, feature) } unless @target_features[target.name]
-        unless @target_plugin_hooks[target.name]
-          set_plugin_hooks(target.name, @config.plugin_hooks.merge(data['plugin_hooks'] || {}))
+        # Add defaults for special 'localhost' target (currently just config and features)
+        if target.name == 'localhost'
+          data = Bolt::Inventory.localhost_defaults(data)
         end
+
+        # Data from inventory
         data['config'] = config_plugin(data['config'])
+        # Data from set_config (make sure to resolve plugins)
+        resolved_target_config = config_plugin(@targets[target.name]['config'] || {})
+        data['config'] = Bolt::Util.deep_merge(data['config'], resolved_target_config)
 
-        # Use Config object to ensure config section is treated consistently with config file
```

Comment:
Is there a reason for removing this?

Reply:
Nope, throughout development on this and the refactors i've done I had a ton of comment notes to myself. I think I deleted this one thinking it was one of mine. Adding it back.